### PR TITLE
Add job_validation.adoc spec for schema and job validation enhancement

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
tendrl-bug-id: Tendrl/specifications#57
Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

### CODE REVIEW

**Comment by TimothyAsir** [comment]
@brainfunked , @r0h4n , @shtripat , @nthomas-redhat Please review

File: specs/job_validation.adoc [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+i.e:- parent_object_name = parameter_name
+For example if the parameter is specified as Volume.brickdetails or Node.Node
+it will validate; where as, sds-validation skips the parameter check if the
+particular parameter is specified without the parent object like Node or
+Nodes or Node[] which may have a single element or a list of elements.
+
+3. Address performance issue
+Currently sds-job-validator validated the integrity of a flow and atoms
+involved for a particular job. If a flow or atom is used multiple time or
+if an atom is used by many flows then the same peace of code will be executed
```
Comment (shtripat) [inline specs/job_validation.adoc]: %s/peace/piece/g

File: specs/job_validation.adoc [hunk 2]
```diff
@@ -0,0 +1,200 @@
+// vim: tw=79
+
+= Definition file and job validation
+
+As there where lot changes in the sds schema, The job-validation module which
+is in the common project needs few changes or enhancement in order to validate
+the sds definition file and the incomming jobs which contains input values.
+
+== Problem description
```
Comment (shtripat) [inline specs/job_validation.adoc:10]: I feel problem description and proposed changes are mixed up here. Move the proposed changes to respective section.

File: specs/job_validation.adoc [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+only one level. Which will not help in case of input parameters have multiple
+sub parameters. For example in case of importNode the parameter can have nested
+input in addition to its regular inputs like Node: [ { fqdn, ...} ],
+Disk [ of devices [of type ...] ] and so on. Sds-job-validator should travel
+into the leaf element in the chain and checks its parameter type recursively
+for every items and checks the whole object.
+
+* Input can be the input from other or already executed atoms
+  Currently it validates only if all the necessary inputs are provided. But
+  as per the recent design change, the input can comes from an already executed
```
Comment (shtripat) [inline specs/job_validation.adoc]: grammar for whole document?

**Comment by cybernth** [comment]
@TimothyAsirJeyasing Instead of  describing diff against the currently available implementation, what I would like to see is a specification for the job validation. This specification needs to include the all the details about the job valiadtion use case.

**Review by cybernth — CHANGES_REQUESTED** [review]
Please make changes as mentioned in https://github.com/Tendrl/specifications/pull/66#issuecomment-267263465

File: specs/job_validation.adoc (hunk 2 above)
Comment (TimothyAsir) [reply specs/job_validation.adoc:10]: Done

File: specs/job_validation.adoc (hunk 3 above)
Comment (TimothyAsir) [reply specs/job_validation.adoc]: Done

**Pull request closed without merging**